apps/aevm/src/aevm_eeevm_state.erl

```diff
@@ -237,10 +237,11 @@ do_revert(Us0, Us1, State0) ->
     case vm_version(State0) of
         ?AEVM_01_Sophia_01 ->
             try
-		%% Get the string then convert it to output binary format.
-                {String, State1} = aevm_eeevm_memory:get_area(Us0, Us1, State0),
-		Out = aeso_data:to_binary(String, 0),
-                set_out(Out, State1)
+                %% In Sophia Us1 is a pointer to the actual value.
+                %% The type of the value is always string.
+                HeapValue = aeso_data:heap_value(maps(State0), Us1, mem(State0)),
+                {ok, Out} = aeso_data:heap_to_binary(string, get_store(State0), HeapValue),
```

Comment:
I am not convinced by this change:
* Can [`aeso_data:heap_to_binary`](https://github.com/aeternity/epoch/blob/21a4be0096fc87bc7d2f0f49e9918ec085a84629/apps/aevm/src/aevm_eeevm_state.erl#L260) return something different from `{ok, _}` or fail?
* How is pointer `Us1` looked up in memory? Is the word at that address considered as 0 if unallocated? (Not sure what to expect.)
* Does this modified semantics allow charging cost of REVERT before performing operation? It seems no to me.

Reply:
You could look up the length of the string before converting to binary and check/charge gas based on the length.

Reply:
The new implementation has exactly the same semantics as before, with the difference that the steps are not expressed in the byte code.

Before, the byte code followed the pointer, found out the size, and took the proper offset/size from memory. Now, it follows the pointer and grabs the memory in the vm instead.

For Sophia, this difference in approach is already used for getting arguments and return values for contract calls/primops.

I just wanted to align the way stuff is happening for Sophia, and as a side effect remove a few byte code instructions. When reading the code, I thought there was a mistake when I first read the code, since few things operate this way in Sophia, so I wanted future readers to not have to do this. This apparently failed ;-)

No big deal, so I can drop the optimization if it feels insecure.

Reply:
> ...
> For Sophia, this difference in approach is already used for getting arguments and return values for contract calls/primops.
> 
> I just wanted to align the way stuff is happening for Sophia ...

Thanks for providing more context. I did not check the other usages (I see related changes in recent master).

Reply:
> You could look up the length of the string before converting to binary and check/charge gas based on the length.

@UlfNorell Good point. Affecting consensus though I expect.